### ISSUE TITLE
False positive of Unable to resolve path to module 'typescript-eslint'  import/no-unresolved

### ISSUE THREAD

**Issue by jimmykane** [import-js/eslint-plugin-import#3187]
Hi there, 

just setted up a new project and new eslint .  

Here is the lint config. 

```
import js from '@eslint/js';
import globals from 'globals';
import reactHooks from 'eslint-plugin-react-hooks';
import reactRefresh from 'eslint-plugin-react-refresh';
import tseslint from 'typescript-eslint';
import json from '@eslint/json';
import markdown from '@eslint/markdown';
import css from '@eslint/css';
import { defineConfig } from 'eslint/config';
import react from 'eslint-plugin-react-x';
import reactDom from 'eslint-plugin-react-dom';
import importPlugin from 'eslint-plugin-import';

export default defineConfig([
    { ignores: ['dist', 'node_modules', 'package-lock.json'] }, 
    {
        files: ['**/*.{js,mjs,cjs,ts,jsx,tsx}'],
        languageOptions: {
            ecmaVersion: 'latest',
            globals: globals.browser
        },
        plugins: {
            'react-hooks': reactHooks,
            'react-refresh': reactRefresh
        },
        rules: {
            'react-refresh/only-export-components': [
                'warn',
                { allowConstantExport: true }
            ],
            'react-x/no-class-component': 'error',
            'react-x/no-unstable-context-value': 'error',
            'import/order': [
                'error',
                {
                    groups: ['builtin', 'external', 'internal'],
                    pathGroups: [
                        {
                            pattern: '{react,react-dom/**}',
                            group: 'external',
                            position: 'before'
                        }
                    ],
                    pathGroupsExcludedImportTypes: ['react']
                }
            ]
        },
        extends: [
            js.configs.recommended,
            tseslint.configs.recommended,
            react.configs.recommended,
            reactDom.configs.recommended,
            importPlugin.flatConfigs.recommended,
            importPlugin.flatConfigs.typescript
        ]
    },
    {
        files: ['**/*.{ts}'],
        extends: [
            tseslint.configs.strictTypeChecked]
    },
    {
        files: ['**/*.json'],
        ignores: ['dist', 'tsconfig*.*.json'],
        plugins: { json },
        language: 'json/json',
        extends: ['json/recommended']
    },
    {
        files: ['**/*.jsonc', 'tsconfig*.*.json'],
        plugins: { json },
        language: 'json/jsonc',
        extends: ['json/recommended']
    },
    { files: ['**/*.json5'], plugins: { json }, language: 'json/json5', extends: ['json/recommended'] },
    { files: ['**/*.md'], plugins: { markdown }, language: 'markdown/commonmark', extends: ['markdown/recommended'] },
    { files: ['**/*.css'], plugins: { css }, language: 'css/css', extends: ['css/recommended'] }
]);
```

However what this produces is : 

```
  5:22  error  Unable to resolve path to module 'typescript-eslint'  import/no-unresolved
  9:30  error  Unable to resolve path to module 'eslint/config'      import/no-unresolved
```

Strange as these modules are enabled and followed the readme correctly  

Any tips?

**Comment by JounQin** [comment]
Please open issue with a runnable reproduction.

**Comment by jimmykane** [comment]
@JounQin  sorry but what more is needed? Run the above and the error comes from the same config file 

```
eslint.config.mjs errors -> 

  5:22  error  Unable to resolve path to module 'typescript-eslint'  import/no-unresolved
  9:30  error  Unable to resolve path to module 'eslint/config'      import/no-unresolved
```

**Comment by JounQin** [comment]
You should provide a GitHub repository so that others can help debugging easily. And it's just duplicate of #1810.

You can try https://github.com/un-ts/eslint-plugin-import-x#import-xresolver-next with `createNodeResolver` instead.
